src/ast/mod.rs

```diff
@@ -8433,6 +8430,31 @@ impl fmt::Display for SessionParamValue {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum SetVariableScope {
+    /// LOCAL (transaction or function) scope
+    Local,
+    /// SESSION scope
+    Session,
+    /// MySQL-specific `GLOBAL` scope
+    Global,
+    /// Not specified; usually implies SESSION
+    None,
+}
+
+impl fmt::Display for SetVariableScope {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Local => write!(f, " LOCAL"),
+            Self::Session => write!(f, " SESSION"),
+            Self::Global => write!(f, " GLOBAL"),
+            Self::None => write!(f, ""),
```

Comment:
does it work the same to define scope as an optional field (`scope: Option<SetVariableScope>`)? in order to represent `None` by convention?